pkgs/applications/networking/instant-messengers/psi-plus/default.nix

```diff
@@ -21,20 +43,30 @@ mkDerivation rec {
   };
 
   cmakeFlags = [
-    "-DENABLE_PLUGINS=ON"
-  ];
-
-  nativeBuildInputs = [ cmake qttools ];
-
-  buildInputs = [
-    qtbase qtmultimedia qtx11extras qtwebengine
-    libidn qca-qt5 libXScrnSaver hunspell
-    libsecret libgcrypt libotr html-tidy libgpgerror libsignal-protocol-c
-    usrsctp
-  ] ++ lib.optionals voiceMessagesSupport [
-    gst_all_1.gst-plugins-base
-    gst_all_1.gst-plugins-good
-  ];
+    "-DCHAT_TYPE=${chatType}"
+    "-DENABLE_PLUGINS=${if enablePlugins then "ON" else "OFF"}"
+    "-DBUILD_PSIMEDIA=${if enablePsiMedia then "ON" else "OFF"}"
+  ] ++ (
+    assert builtins.all builtins.isString extraCmakeFlags; extraCmakeFlags
+  );
+
+  nativeBuildInputs =
+    [ cmake qttools ]
+    ++ lib.optional enablePsiMedia pkg-config;
+
+  buildInputs =
+    [
+      qtbase qtmultimedia qtx11extras
+      libidn qca-qt5 libXScrnSaver hunspell
+      libsecret libgcrypt libotr html-tidy libgpgerror libsignal-protocol-c
+      usrsctp
+    ]
+    ++ lib.optionals voiceMessagesSupport [
+      gst_all_1.gst-plugins-base
+      gst_all_1.gst-plugins-good
+    ]
+    ++ lib.optional (chatType == "webkit") qtwebkit
+    ++ lib.optional (chatType == "webengine") qtwebengine;
```

Comment:
```suggestion
  buildInputs = [
    qtbase qtmultimedia qtx11extras
    libidn qca-qt5 libXScrnSaver hunspell
    libsecret libgcrypt libotr html-tidy libgpgerror libsignal-protocol-c
    usrsctp
  ] ++ lib.optionals voiceMessagesSupport [
    gst_all_1.gst-plugins-base
    gst_all_1.gst-plugins-good
  ] ++ lib.optional (chatType == "webkit") qtwebkit
  ++ lib.optional (chatType == "webengine") qtwebengine;
```

Reply:
@SuperSandro2000 are you following some style guide or is it a personal preference?

My original intention was to use a block indentation and every entry on its own line. If you take the last line and hide others ones:

``` nix
buildInputs = [
# ...
++ lib.optional (chatType == "webengine") qtwebengine;
```

They are on the same indentation level which looks a bit weird to me.

Reply:
Yeah, maybe it needs to be intended, too.

Reply:
I would decline this one. In current form it’s being patch-friendly. Any new entry is added independently.

If you take this one even with indentation:
``` nix
buildInputs = [
  # ...
] ++ lib.optional (chatType == "webkit") qtwebkit
  ++ lib.optional (chatType == "webengine") qtwebengine;
```

Imagine adding another list:

``` nix
buildInputs = [
  # ...
] ++ lib.optional (chatType == "webkit") qtwebkit
  ++ lib.optional (chatType == "webengine") qtwebengine
  ++ lib.optionals foo [
  # ...
];
```

This would look just weird. But with original style it looks more or less okay:

``` nix
buildInputs =
  [
    # ...
  ] 
  ++ lib.optional (chatType == "webkit") qtwebkit
  ++ lib.optional (chatType == "webengine") qtwebengine
  ++ lib.optionals foo [
    # ...
  ];
```